### PR TITLE
Add dialog for creating and editing banners AB#12665

### PR DESCRIPTION
**Pull request by MikeLyttle**
# Implements [AB#12665](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/12665)

## Description

- adds dialogue for creating and editing banners [AB#12665](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/12665)
- adds a rich text editor component using Blazored.TextEditor
- adds missing reducers for updating communications

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [ ] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [x] Not Required

### UI Changes

YES

### Browsers Tested

-   [x] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

-   Fulfills Work Item Requirements

    -   The changes meet/implement story's acceptance criteria. Fixes identified problems if bug.
    -   Changes not directly related to the task requirements need to be documented on the PR.

-   Compilation / Tests

    -   The changes do not break compilation and/or unit or functional tests; unless other item addresses them specifically.

-   Logic Problems / Functional Behaviour

    -   The changes work as intended and do not introduce problems.

-   Performance

    -   The changes do not introduce obvious performance issues.

-   Documented Standards

    -   The changes adhere to the team's documented [Coding Standards](https://github.com/bcgov/healthgateway/wiki/standards).

-   Readability / Maintainability
    -   The changes can be easily understood/read and allow for future enhancements without major refactoring. Readability is preferred over "clever code".
    -   Reasoning for changes needs to be clearly articulated. Disagreements should be arbitrated by a third developer.

### CODE REVIEW

File: Apps/Admin/Client/Components/HgRichTextEditor.razor.css [hunk 1]
```diff
@@ -0,0 +1,22 @@
+::deep .ql-toolbar.ql-snow,
```
Comment (pagustinqt20) [inline Apps/Admin/Client/Components/HgRichTextEditor.razor.css:1]: what is "::"

File: Apps/Admin/Client/Pages/CommunicationsPage.razor.cs [hunk 2]
```diff
@@ -31,37 +36,135 @@ public partial class CommunicationsPage : FluxorComponent
...
+        {
+            0 => CommunicationType.Banner,
+            1 => CommunicationType.InApp,
+            _ => null,
+        };
+
+    private string? SelectedBannerName =>
+        this.SelectedCommunicationType switch
+        {
+            CommunicationType.Banner => "Global",
```
Comment (pagustinqt20) [inline Apps/Admin/Client/Pages/CommunicationsPage.razor.cs:87]: is this the "Banner Posts" tab in old admin?

File: Apps/Admin/Client/Store/Communications/CommunicationsEffects.cs [hunk 3]
```diff
@@ -112,7 +112,7 @@ public async Task HandleUpdateAction(CommunicationsActions.UpdateAction action,
         if (response.IsSuccessStatusCode && response.Content != null && response.Content.ResultStatus == ResultType.Success)
```
Comment (pagustinqt20) [inline Apps/Admin/Client/Store/Communications/CommunicationsEffects.cs:112]: Do we need to ApiResponse? or just try and catch?

File: Apps/Admin/Client/Store/Communications/CommunicationsReducers.cs [hunk 4]
```diff
@@ -161,6 +161,84 @@ public static CommunicationsState ReduceAddFailAction(CommunicationsState state,
...
+    [ReducerMethod(typeof(CommunicationsActions.UpdateAction))]
+    public static CommunicationsState ReduceUpdateAction(CommunicationsState state)
+    {
+        return state with
+        {
+            Update = state.Update with
+            {
+                IsLoading = true,
+            },
+            IsLoading = true,
```
Comment (pagustinqt20) [inline Apps/Admin/Client/Store/Communications/CommunicationsReducers.cs:178]: why two IsLoading?

**Review by pagustinqt20 — APPROVED** [review]

File: Apps/Admin/Client/Store/Communications/CommunicationsEffects.cs (hunk 3 above)
Comment (MikeLyttle) [reply Apps/Admin/Client/Store/Communications/CommunicationsEffects.cs:112]: Hmm, I'll have to think about that.  There is very little difference functionally.

File: Apps/Admin/Client/Store/Communications/CommunicationsReducers.cs (hunk 4 above)
Comment (MikeLyttle) [reply Apps/Admin/Client/Store/Communications/CommunicationsReducers.cs:178]: For the communications state, I experimented with tracking both the state of individual requests (Load, Add, Update, Delete) as well as an overall state, so consumers can check if any requests are loading or have encountered an error, or if a specific request is loading or has encountered an error.

I think I ended up retrieving the global `IsLoading` state along with individual `Error` states.

**Pull request merged**